Firestore/core/src/firebase/firestore/remote/grpc_stream_operation.cc

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/remote/grpc_stream_operation.h"
+
+#include "Firestore/core/src/firebase/firestore/remote/grpc_stream.h"
+#include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+using util::AsyncQueue;
+
+GrpcStreamCompletion::GrpcStreamCompletion(AsyncQueue* firestore_queue,
+                                           Completion&& completion)
+    : completion_{(completion)}, firestore_queue_{firestore_queue} {
```

Comment:
`std::move(completion)`?

Also the argument order doesn't match the initialization order which makes me do a double take.